### PR TITLE
Take git_protocol from gh config

### PR DESCRIPTION
**Pull request by dlvhdr**

### CODE REVIEW

**Comment by ldelossa** [comment]
@dlvhdr can we avoid the large formatter changes in this PR? Its creating a big diff.

**Review by ldelossa — APPROVED** [review]
LGTM! Thanks for this and good idea.

**Comment by ldelossa** [comment]
@dlvhdr approved, but theres a merge conflict

**Comment by dlvhdr** [comment]
@ldelossa I fixed the PR a bit. After testing it, seems like I needed to not parse it gh response as JSON.
I also removed the newline returned from the command.
Now everything works.
Feel free to land, I think I don't have the permission to.

File: lua/litee/gh/ghcli/init.lua [hunk 1]
```diff
@@ -680,4 +680,14 @@ function M.get_check_runs_by_suite(suite_id, on_read)
     async_request(args, on_read, true)
 end
 
+function M.get_git_protocol()
+	local cmd = [[gh config get git_protocol]]
```
Comment (ldelossa) [inline lua/litee/gh/ghcli/init.lua:684]: @dlvhdr are you noticing odd spacing?

**Comment by dlvhdr** [comment]
@ldelossa fixed

**Pull request merged**